File: research/explorations/bench-realistic/fib.py
import sys, time

def fib(n):
    if n <= 1:
        return n
    return fib(n - 1) + fib(n - 2)

N = int(sys.argv[1]) if len(sys.argv) > 1 else 25
# warmup
for _ in range(5):
    fib(N)
iters = 100
start = time.monotonic_ns()
for _ in range(iters):
    r = fib(N)
elapsed = time.monotonic_ns() - start
print(f"result:     {r}")
print(f"iterations: {iters}")
print(f"total:      {elapsed / 1e6:.2f}ms")
print(f"per call:   {elapsed // iters}ns")

File: research/explorations/bench-realistic/fib.js
function fib(n) {
    if (n <= 1) return n;
    return fib(n - 1) + fib(n - 2);
}

const N = parseInt(process.argv[2]) || 25;
// warmup
for (let i = 0; i < 100; i++) fib(N);
const iters = 1000;
const start = process.hrtime.bigint();
let r;
for (let i = 0; i < iters; i++) r = fib(N);
const elapsed = Number(process.hrtime.bigint() - start);
console.log(`result:     ${r}`);
console.log(`iterations: ${iters}`);
console.log(`total:      ${(elapsed / 1e6).toFixed(2)}ms`);
console.log(`per call:   ${Math.floor(elapsed / iters)}ns`);
